### ISSUE TITLE
Migration to latest test framework versions

### ISSUE THREAD

**Issue by bkimminich** [juice-shop/juice-shop#164]
Migrate all unit and integration tests to
- [x] **jasmine 2.x**
- [x] karma-jasmine 1.x
- [x] frisby 2.x
##

**Comment by bkimminich** [comment]
Reverted attempted migration due to high effort and unfinished implementation of `frisby` v2. ~~Also `jasmine-node` still has no official release of their v2.~~

Right now stability of v1 seems to outweigh potential benefits of v2. Migration is postponed until at least finaly v2 release of `frisby`.

(depends on https://github.com/vlucas/frisby/issues/316)

Migrating ~~the client-side unit tests (`karma`, `jasmine`) and~~ the e2e suite (`protractor`, `jasmine`) to latest did not cause any immediate problems or extra efforts.

**Comment by bkimminich** [comment]
~~Side effect: Current workaround for `stryker-mutator` using `coverageAnalysis: 'all'` instead of `'perTest'` can only be removed by migrating to `karma-jasmine 1.x`. See https://gitter.im/stryker-mutator/stryker?at=596475200de4d2545e15e275~~ _Fixed by migrating to Jasmine 2!_

**Comment by bkimminich** [comment]
Frisby v2 is in Alpha now, so it could actually be tried out at least: https://github.com/vlucas/frisby/tree/2.0.0-alpha.3

**Comment by bkimminich** [comment]
`frisby` migration currently blocked by https://github.com/vlucas/frisby/issues/364

**Comment by bkimminich** [comment]
https://github.com/vlucas/frisby/issues/364 has been resolved, unblocking this issue.

**Comment by bkimminich** [comment]
Back into _blocked_ state from another issue in frisby v2: https://github.com/vlucas/frisby/issues/365

**Comment by bkimminich** [comment]
Unblocked as acceptable workaround exists: https://github.com/vlucas/frisby/issues/365#issuecomment-318050352

**Comment by bkimminich** [comment]
Small leftover frisby2 issues remain as well as jasmine-node still using Jasmine 1.3 for running frisby.

**Comment by bkimminich** [comment]
Migrated frisby2 tests to run with Jest instead of outdated jasmine-node.
